util.go

```diff
@@ -37,3 +40,44 @@ func prettyPrint(data interface{}) {
 	dbg, _ := json.MarshalIndent(data, "", "  ")
 	fmt.Println(string(dbg))
 }
+
+// GetDefaultRelease attempts to guess a default release.
+func GetDefaultRelease() string {
```

Comment:
If calling this when creating `Event`, wouldn't that mean that a service which suddenly generates a large number of events, will spin up `n` number of `git rev-parse HEAD` processes, making it kind of a fork bomb? If so, I feel like it might be better to setup some sort of situation where the call of this function is locked from sharing, and the result cached.

Reply:
> If so inclined, please see the note I left on the `GetDefaultRelease()` function declaration.

Howdy!
Thanks for giving a look at this pr! Much appreciated!

When I read your comment, at first glance, I didn't think twice, this certainly would blow, like a fork bomb ! 😂

But, the function `GetDefaultRelease` is initialized on `Event`'s on test files only. I imagine this could be an exception for calling `GetDefaultRelease`?

I think that in a normal situation, `GetDefaultRelease` is called only once in `NewClient` on main.go.

"[...] some sort of situation where the call of this function is locked from sharing, [...]"
I am not sure if I understood, by locking, you mean not exporting `GetDefaultRelease`? I didn't export at first, but wasn't able to call this on the test cases, since they are in different packages.

Feel free to point anything else you find strange in this pr!

Reply:
> Howdy!
> Thanks for giving a look at this pr! Much appreciated!

You're very welcome.

> I think that in a normal situation, `GetDefaultRelease` is called only once in `NewClient` on main.go.

You're right, I'm not sure where my mind was with that. I retract.

Reply:
Happy to see the positive engagement here :)

- [x] Re: how often this is called;

  I think we are fine for now. `GetDefaultRelease` is called once per `NewClient` call, and we call `NewClient` once when the SDK is initialized with `Init`, which should be only once in the lifetime of a program. If this ever becomes a problem we can cache the result, but we don't need that to land this PR.

- [ ] Re: exporting this name;

  Please let's not export it. It is easy enough to export it later, it is hard to remove something later. SDK users don't really need to call this function themselves. We can deal with tests, see other comment.